pkg/security/tests/dentry_test.go

```diff
@@ -310,3 +311,109 @@ func TestDentryRmdir(t *testing.T) {
 		}
 	}
 }
+func createOverlayLayer(t *testing.T, test *testModule, name string) string {
+	p, _, err := test.Path(name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	os.MkdirAll(p, os.ModePerm)
+
+	return p
+}
+
+func createOverlayLayers(t *testing.T, test *testModule) (string, string, string, string) {
+	return createOverlayLayer(t, test, "lower"),
+		createOverlayLayer(t, test, "upper"),
+		createOverlayLayer(t, test, "workdir"),
+		createOverlayLayer(t, test, "merged")
+}
+
+func TestDentryOverlay(t *testing.T) {
+	if testEnvironment == DockerEnvironment {
+		t.Skip()
+	}
+
+	rule := &rules.RuleDefinition{
+		ID:         "test_rule",
+		Expression: `open.filename == "{{.Root}}/merged/kiki.txt"`,
+	}
+
+	test, err := newTestModule(nil, []*rules.RuleDefinition{rule}, testOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	testLower, testUpper, testWordir, testMerged := createOverlayLayers(t, test)
```

Comment:
nit: delete these directories at the end of the test.

Reply:
I think this is done by this line
https://github.com/DataDog/datadog-agent/blob/master/pkg/security/tests/setup_test.go#L398 ?

Reply:
Thanks.